mono/mini/mini-exceptions.c

```diff
@@ -1560,18 +1560,17 @@ mono_handle_exception_internal (MonoContext *ctx, gpointer obj, gboolean resume,
 			gboolean unhandled = FALSE;
 
 			/*
-			 * The exceptions caught by the mono_runtime_invoke () calls in mono_async_invoke () needs to be treated as
-			 * unhandled (#669836).
-			 * FIXME: The check below is hackish, but its hard to distinguish these runtime invoke calls from others
-			 * in the runtime.
+			 * The exceptions caught by the mono_runtime_invoke () calls
+			 * in the threadpool needs to be treated as unhandled (#669836).
```

Comment:
@kumpera also as explained in this comment, what we want is special case the mono_runtime_invoke in the threadpool, and because we always call mono_defaults.threadpool_perform_wait_callback_method, it make sense to have it hardcoded in the following check